### PR TITLE
Fix typo and improve readability

### PR DESCRIPTION
**Pull request by charles1046**
I don't think combining multiple booleans together and using reverse sign are good for readability.

I recommend doing separated returns for every condition and specifying explicitly what you are going to compare.

### CODE REVIEW

File: src/http/body.c [hunk 1]
```diff
@@ -16,20 +16,19 @@ struct Body {
...
+#define SPAIR(node) ((SPair*)(node)->value)
 
 // If you want to implement Content-Type please refer RFC 2616 7.1
 
-static inline _Bool is_sepported_type(const char* str) {
-	if(!str)
-		return false;
-	return !strncmp(str, "application/x-www-form-urlencoded", 34);
+static inline _Bool is_supported_type(const char* str) {
+	return !strncmp(str, "application/x-www-form-urlencoded", 33);
```
Comment (25077667) [inline src/http/body.c:24]: What if str is a NULL ptr?

File: src/http/header.c [hunk 2]
```diff
@@ -94,17 +94,22 @@ void header_insert_move(Header* restrict h, SSPair* restrict p) {
 }
 
 void* header_get(const Header* restrict h, const char* restrict key) {
-	if(unlikely(!h || !key || *key))
-		return NULL;
+	if(unlikely(h == NULL))
```
Comment (25077667) [inline src/http/header.c:97]: Well, I think the original condition is okay.

File: src/http/query.c [hunk 3]
```diff
@@ -43,11 +43,17 @@ Query* query_parser(char* buffer) {
 }
 
 void* query_get(const Query* restrict q, const char* restrict key) {
-	if(unlikely(!q || !key || *key))
-		return NULL;
+	if(unlikely(q == NULL))
```
Comment (25077667) [inline src/http/query.c:46]: Also here.

File: src/http/query.c [hunk 4]
```diff
@@ -43,11 +43,17 @@ Query* query_parser(char* buffer) {
...
+
+	if(unlikely(key == NULL))
+		return NULL;  // no key
+
+	if(unlikely(*key == '\0'))
+		return NULL;  // empty key
 
 	Node* cur = q->head;
-	while(cur && strcmp(SPAIR(cur)->key, key))
+	while(cur && strcmp(SPAIR(cur)->key, key) != 0)
```
Comment (25077667) [inline src/http/query.c:56]: It is an unessential condition statement.
We know the strcmp's return value: https://man7.org/linux/man-pages/man3/strcmp.3.html

File: src/server.c [hunk 5]
```diff
@@ -114,19 +114,19 @@ static void serverDelFd(Server* server, int fd) {
 }
 
 static inline void handle(Server* server, int fd, struct sockaddr_in* addr) {
-	int nread = 0;
 	char buff[20480];
+	long n_read = recv(fd, buff, sizeof(buff), 0);
```
Comment (25077667) [inline src/server.c:118]: Good!

**Review by 25077667 — APPROVED** [review]
Please refer to those comments.

File: src/http/body.c (hunk 1 above)
Comment (charles1046) [reply src/http/body.c:24]: The function will be called if content_type is not null.
Refer to line 29.
I recommend refactor that condition or keep this change.

File: src/http/header.c (hunk 2 above)
Comment (charles1046) [reply src/http/header.c:97]: The third condition should be !(*key) then.
Combining booleans together will easily cause bug like this and hard to find.

File: src/http/query.c (hunk 3 above)
Comment (charles1046) [reply src/http/query.c:46]: Also here.

File: src/http/query.c (hunk 4 above)
Comment (charles1046) [reply src/http/query.c:56]: Here should be like src/http/header.c line 107.
I forgot to change.

File: src/server.c (hunk 5 above)
Comment (charles1046) [reply src/server.c:118]: Thanks.

File: src/http/header.c (hunk 2 above)
Comment (25077667) [reply src/http/header.c:97]: If key is NULL, it will trigger the short cut and enter this segment.

Comment (charles1046) [reply src/http/header.c:97]: If `key` is not `NULL` and it's not empty string (like `"\0"`), `*key` will not be 0.
Which means the condition is true, the segment is triggered, and it returns NULL.

Comment (25077667) [reply src/http/header.c:97]: Obviously, yes.

Comment (25077667) [reply src/http/header.c:97]: Oh yes, it is a bug.

**Pull request merged**